src/main/java/com/google/devtools/build/lib/skyframe/TargetCycleReporter.java

```diff
@@ -13,49 +13,73 @@
 import com.google.devtools.build.lib.events.ExtendedEventHandler;
 import com.google.devtools.build.lib.packages.PackageGroup;
 import com.google.devtools.build.lib.packages.Target;
 import com.google.devtools.build.lib.pkgcache.PackageProvider;
+import com.google.devtools.build.lib.skyframe.AspectValueKey.AspectKey;
 import com.google.devtools.build.skyframe.CycleInfo;
 import com.google.devtools.build.skyframe.SkyKey;
 import java.util.List;
 
 /**
- * Reports cycles between {@link TransitiveTargetValue}s. These indicates cycles between targets
- * (e.g. '//a:foo' depends on '//b:bar' and '//b:bar' depends on '//a:foo').
+ * Reports cycles between targets. These may be in the form of {@link ConfiguredTargetValue}s or
+ * {@link TransitiveTargetValue}s.
  */
-class TransitiveTargetCycleReporter extends AbstractLabelCycleReporter {
-  private static final Predicate<SkyKey> IS_SUPPORTED_SKY_KEY =
+class TargetCycleReporter extends AbstractLabelCycleReporter {
+
+  private static final Predicate<SkyKey> SUPPORTED_SKY_KEY =
       Predicates.or(
+          SkyFunctions.isSkyFunction(SkyFunctions.CONFIGURED_TARGET),
+          SkyFunctions.isSkyFunction(SkyFunctions.ASPECT),
+          SkyFunctions.isSkyFunction(SkyFunctions.LOAD_STARLARK_ASPECT),
           SkyFunctions.isSkyFunction(SkyFunctions.TRANSITIVE_TARGET),
           SkyFunctions.isSkyFunction(SkyFunctions.PREPARE_ANALYSIS_PHASE));
 
-  TransitiveTargetCycleReporter(PackageProvider packageProvider) {
+  TargetCycleReporter(PackageProvider packageProvider) {
     super(packageProvider);
   }
 
+  @Override
+  protected boolean shouldSkipOnPathToCycle(SkyKey key) {
+    return SkyFunctions.PREPARE_ANALYSIS_PHASE.equals(key.functionName());
+  }
+
   @Override
   protected boolean canReportCycle(SkyKey topLevelKey, CycleInfo cycleInfo) {
-    return Iterables.all(Iterables.concat(ImmutableList.of(topLevelKey),
-        cycleInfo.getPathToCycle(), cycleInfo.getCycle()),
-        IS_SUPPORTED_SKY_KEY);
+    return SUPPORTED_SKY_KEY.apply(topLevelKey)
+        && cycleInfo.getPathToCycle().stream().allMatch(SUPPORTED_SKY_KEY)
+        && cycleInfo.getCycle().stream().allMatch(SUPPORTED_SKY_KEY);
   }
 
   @Override
-  protected Label getLabel(SkyKey key) {
-    return ((TransitiveTargetKey) key).getLabel();
+  public String prettyPrint(SkyKey key) {
+    if (SkyFunctions.isSkyFunction(SkyFunctions.CONFIGURED_TARGET).apply(key)) {
+      return ((ConfiguredTargetKey) key.argument()).prettyPrint();
+    } else if (SkyFunctions.isSkyFunction(SkyFunctions.ASPECT).apply(key)) {
+      return ((AspectKey) key.argument()).prettyPrint();
+    } else if (key instanceof AspectValueKey) {
+      return ((AspectValueKey) key).getDescription();
+    } else {
+      return getLabel(key).toString();
+    }
   }
 
   @Override
-  protected boolean shouldSkipOnPathToCycle(SkyKey key) {
-    return SkyFunctions.PREPARE_ANALYSIS_PHASE.equals(key.functionName());
+  public Label getLabel(SkyKey key) {
```

Comment:
It's been a while since I looked at this code. Why's this a different set of key types than prettyPrint?

Reply:
Both the CT and aspect keys implement `ActionLookupValue`, which can be handled generically here but want to be pretty-printed differently.